app/root.tsx

```diff
@@ -25,19 +33,34 @@ export const meta: MetaFunction = () => ({
   viewport: 'width=device-width,initial-scale=1',
 });
 
-const App = () => (
-  <html lang="en">
-    <head>
-      <Meta />
-      <Links />
-    </head>
-    <body>
-      <Outlet />
-      <ScrollRestoration />
-      <Scripts />
-      <LiveReload />
-    </body>
-  </html>
-);
+export async function loader() {
+  return json({
+    ENV: envObj,
+  });
+}
+
+const App = () => {
+  const data = useLoaderData<typeof loader>();
+
+  return (
+    <html lang="en">
+      <head>
+        <Meta />
+        <Links />
+      </head>
+      <body>
+        <Outlet />
+        <script
+          dangerouslySetInnerHTML={{
+            __html: `window.ENV = ${JSON.stringify(data.ENV)}`,
+          }}
+        />
```

Comment:
let's find another way to do this.

Reply:
we are referring the docs
https://remix.run/docs/en/1.14.0/guides/envvars,
i think we can keep it right?